tests/unit/TestNetRemoteDataStreamingReactors.cxx

```diff
@@ -0,0 +1,67 @@
+
+#include "TestNetRemoteDataStreamingReactors.hxx"
+
+using namespace Microsoft::Net::Remote::DataStream;
+using namespace Microsoft::Net::Remote::Service;
+using namespace Microsoft::Net::Remote::Test;
+
+DataStreamWriter::DataStreamWriter(NetRemoteDataStreaming::Stub* client, uint32_t numberOfDataBlocksToWrite) :
+    m_numberOfDataBlocksToWrite(numberOfDataBlocksToWrite)
+{
+    client->async()->DataStreamUpload(&m_clientContext, &m_result, this);
+    StartCall();
+    NextWrite();
+}
+
+void
+DataStreamWriter::OnWriteDone(bool isOk)
+{
+    if (isOk) {
+        NextWrite();
+    } else {
+        StartWritesDone();
+    }
+}
+
+void
+DataStreamWriter::OnDone(const grpc::Status& status)
+{
+    std::unique_lock lock(m_writeStatusGate);
+
+    m_status = status;
+    m_done = true;
+    m_writesDone.notify_one();
+}
+
+grpc::Status
+DataStreamWriter::Await(DataStreamUploadResult* result)
+{
+    std::unique_lock lock(m_writeStatusGate);
+    static constexpr auto timeoutValue = 10s;
+
+    const auto isDone = m_writesDone.wait_for(lock, timeoutValue, [this] {
+        return m_done;
+    });
+
+    if (!isDone) {
+        DataStreamOperationStatus status{};
+        status.set_code(DataStreamOperationStatusCode::DataStreamOperationStatusCodeFailed);
```

Comment:
Would a ne wstatus code like `DataStreamOperationStatusCode::DataStreamOperationStatusTimedOut` make sense here?

Reply:
I think that's a good idea. Probably best to have the status codes be as specific as possible and leave the "failed" code for failures without a clear cause